docs/use-cases/NVIDIA-GPU-passthrough-and-Kata.md

```diff
@@ -1,621 +1,346 @@
+These operands are vital to support the flow of orchestrating pod manifests using NVIDIA GPU runtime classes with GPU passthrough on your nodes.
+Without getting into the details, the most important operands and their responsibilities are:
+- **nvidia-vfio-manager:** Binding discovered NVIDIA GPUs to the `vfio-pci` driver for VFIO passthrough.
+- **nvidia-cc-manager:** Transitioning GPUs into confidential computing (CC) and non-CC mode.
+- **nvidia-kata-manager:** Creating host-side CDI specifications for GPU passthrough (`/var/run/cdi/nvidia.yaml`, containing `kind: nvidia.com/pgpu`).
+- **nvidia-sandbox-device-plugin** (see the repository [NVIDIA/sandbox-device-plugin](https://github.com/NVIDIA/sandbox-device-plugin) for the plugin being utilized on Kata CI runners):
+  - Allocating GPUs during pod deployment.
+  - Discovering NVIDIA GPUs, their capabilities, and advertising these to the Kubernetes control plane (`nvidia.com/gpu` resources will appear as node labels, GPU DeviceIDs will be registered with kubelet) which can in turn be requested by your pod manifests.
+
+To summarize, the GPU operator manages the GPUs on each node, allowing for simple orchestration of pod manifests using Kata Containers.
+Once the cluster with GPU operator and Kata bits is up and running, the end user can schedule Kata NVIDIA GPU workloads, using resource limits and the `kata-qemu-nvidia-gpu` or `kata-qemu-nvidia-gpu-snp` runtime classes, for example:
+```
+apiVersion: v1
+kind: Pod
+...
+spec:
+  ...
+  runtimeClassName: kata-qemu-nvidia-gpu-snp
+  ...
+    resources:
+      limits:
+        "nvidia.com/pgpu": 1
+...
+```
+
+When this happens, kubelet calls into the sandbox device plugin to allocate a GPU.
+The sandbox device plugin returns a GPU device ID and outer runtime CDI annotations to the kubelet.
+The kubelet uses the GPU device ID for tracking of allocated GPUs and includes the CDI annotations in the OCI spec when scheduling the pod through containerd.
+Containerd in turn invokes the Kata runtime with the OCI spec.
+
+### Kata runtime
+The Kata runtime receives the OCI spec with these annotations and then queries kubelet's Pod Resources API using the information from the annotations to acquire the GPU device ID allocated by the sandbox device plugin.
+The runtime thus realizes that, for example, the GPU with ID `GPU0` is requested.
+The runtime then consults the host CDI specifications and determines the device path the GPU is backed by (e.g., `/dev/vfio/devices/vfio0`).
+Then, the runtime resolves the device's PCI BDF (e.g., `0000:01:00.0`).
+
+For NVIDIA GPU TEE workloads, the runtime cold-plugs the GPU by launching QEMU with relevant parameters for device passthrough (e.g., `-device vfio-pci,host=0000:01:00.0,id=gpu0,bus=pcie.0,addr=0x2`).
+The runtime then creates "inner runtime" CDI annotations for workload containers which are sent to kata-agent as part of the create container request.
```

Comment:
I don't think it's the containerd shim that creates the inner runtime, it's the kata agent that asks the nvidia-ctk for it. Someone should double check me here. The agent receives the device IDs from the shim but not enough information on where the devices are exposed in the VM (again, someone correct me if I'm wrong, I could be).

Reply:
I am trying to clarify this right now - please take a look at `createCDIAnnotation` in `src/runtime/virtcontainers/container.go` - this is what I was referring to here. 

I think, in your comment you were referring to the inner CDI specification file